common/src/escrow_accounts.rs

```diff
@@ -104,22 +106,31 @@ pub fn escrow_accounts(
     indexer_address: Address,
     interval: Duration,
     reject_thawing_signers: bool,
-) -> Eventual<EscrowAccounts> {
-    timer(interval).map_with_retry(
-        move |_| async move {
-            get_escrow_accounts(escrow_subgraph, indexer_address, reject_thawing_signers)
-                .await
-                .map_err(|e| e.to_string())
-        },
-        move |err: String| {
-            error!(
-                "Failed to fetch escrow accounts for indexer {:?}: {}",
-                indexer_address, err
-            );
-
-            sleep(interval.div_f32(2.0))
-        },
-    )
+) -> Receiver<EscrowAccounts> {
+    let (tx, rx) = watch::channel(EscrowAccounts::default());
+    tokio::spawn(async move {
+        let mut time_interval = time::interval(interval);
+        time_interval.set_missed_tick_behavior(time::MissedTickBehavior::Skip);
+        loop {
+            time_interval.tick().await;
+            let result =
+                get_escrow_accounts(escrow_subgraph, indexer_address, reject_thawing_signers).await;
+            match result {
+                Ok(accounts) => tx
+                    .send(accounts)
+                    .expect("Failed to update escrow_accounts channel"),
+                Err(err) => {
+                    error!(
+                        "Failed to fetch escrow accounts for indexer {:?}: {}",
+                        indexer_address, err
+                    );
+                    // Sleep for a bit before we retry
+                    sleep(interval.div_f32(2.0)).await;
+                }
+            }
+        }
+    });
+    rx
```

Comment:
Hey, could you do that same thing we talked about that HashSet? I don't want to initialize a `watch::Receiver` with default values. I might need to change other receivers too

Reply:
Sure, I did that first, removed the option cause it was adding many lines, anyways I'll bring it back.

Reply:
Hey don't need to add Options, I prefer them to be loaded while creating.

Reply:
got it✌️